test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,386 @@
+
+         A use statement makes all of the module's visible symbols available to
+         the scope which contains the use statement.  These symbols may then be
+         accessed without the module name prefix.
+
+         In this case, bazBarFoo should store the result of calling modToUse.baz
+         on modToUse.bar and modToUse.foo, which is in this case '28'.
+      */
+      var bazBarFoo = baz(bar, foo);
+      writeln(bazBarFoo);
+    }
+
+
+
+    /* Since the following line doesn't live within a scope that contains a
+       'use' of 'modToUse', it would generate an error if uncommented.  This is
+       because 'foo' cannot be directly referenced, and is not qualified with a
+       module name.
+    */
+    // var twiceFoo = 2 * foo;
+
+
+
+    {
+      var bazBarFoo = baz(bar, foo);
+
+      /* Use statements apply to the entire scope in which they are defined.
+         Even if the use statement occurs after code which would directly refer
+         to its symbols, these references are still valid.  This is similar to
+         other Chapel forms of introducing symbols - for instance, class
+         declaration order does not prevent the first declared class from
+         referring to the second.
+
+         Thus, as in an earlier example, bazBarFoo should store the result of
+         calling modToUse.baz on modToUse.bar and modToUse.foo, which is in this
+         case '28'.
+      */
+      use modToUse;
+
+      writeln(bazBarFoo);
+    }
+
+
+    {
+      /* The symbols provided by a use statement are only considered when the
+         name in question cannot be resolved directly within the local scope.
+         Thus, if another bar were defined here, and an access was attempted,
+         the compiler would find the bar at this scope, rather than
+         modToUse.bar.
+      */
+      var bar = 4.0;
+
+      use modToUse;
+
+      writeln(bar);
+      // Will output the value of the bar defined in scope (which is '4.0'),
+      // rather than the value of modToUse.bar (which is '2')
+    }
+
+
+    var bar = false;
+    {
+      /* If a symbol cannot be resolved directly within the local scope, then
+         the symbols provided by a use statement are considered before the
+         symbols defined outside of the scope where the use statement applies.
+         Thus, if another bar were defined outside of these curly braces, and an
+         access was attempted, the compiler would find the bar from modToUse,
+         rather than the outer bar.  The bar from modToUse is said to be
+         "shadowing" the definition at the outer scope.
+      */
+
+      use modToUse;
+      writeln(bar);
+      // Will output the value of modToUse.bar (which is '2'), rather than the
+      // value of the bar defined outside of this scope (which is 'false')
+    }
+
+    {
+      /* Using nested modules combines accessing a module's symbols explicitly
+         and using a top-level module.  If the parent module(s) have not been
```

Comment:
I don't know, I found the whole paragraph confusing.  To me nesting modules, and using nested modules, seems like pretty natural composition -- they're just other symbols defined within modules that happen to be modules themselves and all the rules compose the way you'd expect.  Maybe this is simply a matter of trying to say too much in the abstract where a blanket statement ("Modules can be nested and compose as you'd expect") followed by specific examples might be more useful?